### DISCUSSION TITLE
Enable shiny-for-python to compete with streamlit

### DISCUSSION THREAD

**Discussion by vnijs** [quarto-dev/quarto-cli#6312] [Feature Requests]
### Description

I have been using shiny-for-python and love it. Howevery, Streamlit currently seems to have the largest market share in this space for Python. Shiny-for-python has many benefits compared to Streamlit (https://shiny.posit.co/py/docs/comp-streamlit.html) but the one thing Streamlit wins on is ease of use. To me, Steamlit feels more like an [interactive document](https://rmarkdown.rstudio.com/lesson-14.html) or [flexdashboard](https://pkgs.rstudio.com/flexdashboard/). Quick and easy but not as flexible or powerful as a full shiny app.

Question: Have you considered combining Quarto + Shiny-for-python as a (more) direct competitor to Streamlit? It could be the gateway for more people in the Python community to get to know Shiny.

Just my 2Cs

**Comment by mcanouil** [comment]
Shinylive for python support is only provided through a Quarto extension, see <https://github.com/quarto-ext/shinylive>.

There is nothing (in theory) blocking you from doing the same thing with other tools.

**Comment by vnijs** [comment]
Thanks for the link by the quarto extension for shinylive. Am I correct in assuming that "There is nothing (in theory) blocking you from doing the same thing with other tools." means, in short, "If you want it, make an extension"?

**Reply by mcanouil** [reply]
Basically yes, this is the point of the extension mechanism, i.e., allowing users to extend how ever they want with whatever they want Quarto.

This does not mean you are completely on your own.
As you can see in the discussions, there are many about making extensions.
